File: lab2 py, c#/lab2.py
# Базовый класс Vehicle
class Vehicle:
    def __init__(self, make, model, year):
        """
        Конструктор базового класса.
        Передача данных: параметры make, model, year передаются при создании объекта
        и сохраняются в атрибуты экземпляра через self.
        """
        self.make = make      # марка
        self.model = model    # модель
        self.year = year      # год выпуска

    def info(self):
        """Метод для вывода информации о транспортном средстве."""
        return f"{self.year} {self.make} {self.model}"


# Подкласс Car — наследует Vehicle
class Car(Vehicle):
    def __init__(self, make, model, year, doors, body_style):
        """
        Конструктор подкласса Car.
        Передача данных: make, model, year передаются в базовый класс Vehicle через super().
        doors и body_style — собственные атрибуты Car.
        """
        super().__init__(make, model, year)  # вызов конструктора базового класса
        self.doors = doors
        self.body_style = body_style

    def info(self):
        """Переопределённый метод (полиморфизм)."""
        return f"{super().info()} | {self.doors} двери | кузов: {self.body_style}"


# Подкласс Truck — наследует Vehicle
class Truck(Vehicle):
    def __init__(self, make, model, year, bed_length, towing_capacity):
        """
        Конструктор подкласса Truck.
        Передача данных: make, model, year передаются в базовый класс Vehicle.
        bed_length и towing_capacity — собственные атрибуты Truck.
        """
        super().__init__(make, model, year)
        self.bed_length = bed_length      # длина кузова
        self.towing_capacity = towing_capacity  # тяговая мощность

    def info(self):
        """Переопределённый метод (полиморфизм)."""
        return f"{super().info()} | кузов: {self.bed_length} м | тяга: {self.towing_capacity} кг"


# Создание объектов и вывод результата
if __name__ == "__main__":
    print("=== Автосалон: объекты и наследование ===\n")

    # Объект базового класса
    vehicle = Vehicle("Toyota", "Camry", 2020)
    print("Базовый класс Vehicle:")
    print(vehicle.info())
    print()

    # Объект подкласса Car
    car = Car("Honda", "Civic", 2022, 4, "седан")
    print("Подкласс Car (наследует Vehicle):")
    print(car.info())
    print(f"  Атрибуты Car: марка={car.make}, модель={car.model}, год={car.year}, двери={car.doors}, кузов={car.body_style}")
    print()

    # Объект подкласса Truck
    truck = Truck("Ford", "F-150", 2023, 2.5, 5000)
    print("Подкласс Truck (наследует Vehicle):")
    print(truck.info())
    print(f"  Атрибуты Truck: марка={truck.make}, модель={truck.model}, год={truck.year}, длина кузова={truck.bed_length} м, тяга={truck.towing_capacity} кг")

File: lab2 py, c#/lab2.cs
using System;

namespace AutoSalon
{
    // Базовый класс Vehicle
    class Vehicle
    {
        public string Make { get; set; }   // марка
        public string Model { get; set; }  // модель
        public int Year { get; set; }      // год выпуска

        // Конструктор базового класса
        public Vehicle(string make, string model, int year)
        {
            // Передача данных: параметры присваиваются свойствам
            Make = make;
            Model = model;
            Year = year;
        }

        // Виртуальный метод (может быть переопределён в подклассах)
        public virtual string Info()
        {
            return $"{Year} {Make} {Model}";
        }
    }

    // Подкласс Car — наследует Vehicle
    class Car : Vehicle
    {
        public int Doors { get; set; }          // количество дверей
        public string BodyStyle { get; set; }   // стиль кузова

        // Конструктор подкласса Car
        public Car(string make, string model, int year, int doors, string bodyStyle)
            : base(make, model, year)   // вызов конструктора базового класса (передача make, model, year)
        {
            // Собственные атрибуты
            Doors = doors;
            BodyStyle = bodyStyle;
        }

        // Переопределённый метод (полиморфизм)
        public override string Info()
        {
            return $"{base.Info()} | {Doors} двери | кузов: {BodyStyle}";
        }
    }

    // Подкласс Truck — наследует Vehicle
    class Truck : Vehicle
    {
        public double BedLength { get; set; }    // длина кузова
        public int TowingCapacity { get; set; }   // тяговая мощность

        // Конструктор подкласса Truck
        public Truck(string make, string model, int year, double bedLength, int towingCapacity)
            : base(make, model, year)   // вызов конструктора базового класса
        {
            BedLength = bedLength;
            TowingCapacity = towingCapacity;
        }

        // Переопределённый метод (полиморфизм)
        public override string Info()
        {
            return $"{base.Info()} | кузов: {BedLength} м | тяга: {TowingCapacity} кг";
        }
    }

    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine("=== Автосалон: объекты и наследование ===\n");

            // Объект базового класса
            Vehicle vehicle = new Vehicle("Toyota", "Camry", 2020);
            Console.WriteLine("Базовый класс Vehicle:");
            Console.WriteLine(vehicle.Info());
            Console.WriteLine();

            // Объект подкласса Car
            Car car = new Car("Honda", "Civic", 2022, 4, "седан");
            Console.WriteLine("Подкласс Car (наследует Vehicle):");
            Console.WriteLine(car.Info());
            Console.WriteLine($"  Атрибуты Car: марка={car.Make}, модель={car.Model}, год={car.Year}, двери={car.Doors}, кузов={car.BodyStyle}");
            Console.WriteLine();

            // Объект подкласса Truck
            Truck truck = new Truck("Ford", "F-150", 2023, 2.5, 5000);
            Console.WriteLine("Подкласс Truck (наследует Vehicle):");
            Console.WriteLine(truck.Info());
            Console.WriteLine($"  Атрибуты Truck: марка={truck.Make}, модель={truck.Model}, год={truck.Year}, длина кузова={truck.BedLength} м, тяга={truck.TowingCapacity} кг");
        }
    }
}